mmdet/models/dense_heads/detr_head.py

```diff
@@ -241,18 +241,36 @@ def forward_single(self, x, img_metas):
         # NOTE following the official DETR repo, non-zero values representing
         # ignored positions, while zero values means valid positions.
         batch_size = x.size(0)
-        input_img_h, input_img_w = img_metas[0]['batch_input_shape']
-        masks = x.new_ones((batch_size, input_img_h, input_img_w))
-        for img_id in range(batch_size):
-            img_h, img_w, _ = img_metas[img_id]['img_shape']
-            masks[img_id, :img_h, :img_w] = 0
+        if torch.onnx.is_in_onnx_export():
+            # Note `img_shape` is not dynamically traceable to ONNX,
+            # sicne the related augmentation was done with numpy under
+            # CPU. Here `img_shape` and 'batch_input_shape' are discarded,
+            # then `masks` is directly created with zeros (valid tag) and
+            # the same spatial shape as `x`.
+            # The difference between torch and exported ONNX model may be
+            # ignored, since the same performance is achieved (e.g.
+            # 40.1 vs 40.1 for DETR)
+            masks = x.new_zeros(*x.size()[-2:]).unsqueeze(0).repeat(
```

Comment:
`repeat` might bring an extra cost, can we just create a tensor with shape [B,h,w]?

Reply:
or considering expand()